plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java

```diff
@@ -0,0 +1,174 @@
+import org.jkiss.dbeaver.ext.generic.model.GenericUtils;
+import org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaModel;
+import org.jkiss.dbeaver.ext.generic.model.meta.GenericMetaObject;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCPreparedStatement;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCResultSet;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCSession;
+import org.jkiss.dbeaver.model.exec.jdbc.JDBCStatement;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCConstants;
+import org.jkiss.dbeaver.model.impl.jdbc.JDBCUtils;
+import org.jkiss.dbeaver.model.struct.rdb.DBSIndexType;
+
+public class CubridMetaModel extends GenericMetaModel {
+	
+	CubridMetaModelDescriptor descriptor;
+
+	public CubridMetaModel()
+	{
+	}
+
+	public GenericMetaObject getMetaObject(String id) {
+		return descriptor == null ? null : descriptor.getObject(id);
+	}
+
+	public CubridUser createCubridUserImpl(CubridObjectContainer container, String name, String comment) {
+		return new CubridUser(container, name, comment);
+	}
+
+	public JDBCStatement prepareCubridUserLoadStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer container) throws SQLException {
+		String sql= "select * from db_user";
+		final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+		return dbStat;
+	}
+
+	public JDBCStatement prepareTableLoadStatement(@NotNull JDBCSession session)
+		throws SQLException {
+		String sql= "select a.*, case when class_type = 'CLASS' then 'TABLE' \r\n"
+				+ "when class_type = 'VCLASS' then 'VIEW' end as TABLE_TYPE, \r\n"
+				+ "b.current_val from db_class a LEFT JOIN db_serial b on \r\n"
+				+ "a.class_name = b.class_name where a.is_system_class='NO'";
+		final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+		return dbStat;
+	}
+
+	public JDBCStatement prepareSystemTableLoadStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner, @Nullable CubridTableBase object, @Nullable String objectName)
+		throws SQLException {
+		String sql= "select *, class_name as TABLE_NAME, case when class_type = 'CLASS' \r\n"
+	    		+ "then 'TABLE' end as TABLE_TYPE from db_class\r\n"
+	       		+ "where class_type = 'CLASS' \r\n"
+	       		+ "and is_system_class = 'YES'";
+		final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+		return dbStat;
+	}
+
+	public JDBCStatement prepareSystemViewLoadStatement(@NotNull JDBCSession session, @NotNull CubridObjectContainer owner, @Nullable CubridTableBase object, @Nullable String objectName)
+		throws SQLException {
+		String sql= "select *, case when class_type = 'VCLASS' \r\n"
+				+ "then 'VIEW' end as TABLE_TYPE,\r\n"
+				+ "class_name as TABLE_NAME from db_class\r\n"
+				+ "where class_type='VCLASS'\r\n"
+				+ "and is_system_class='YES'";
+		final JDBCPreparedStatement dbStat = session.prepareStatement(sql);
+		return dbStat;
+	}
+
+	public boolean isNewVersion(JDBCSession session) {

```

Comment:
You can't use here "new version" name here. It is new now, but you need to think about the future in the present. A few years later this version will be not new, but this method will have such a name.

So, please change it to "isDatabaseVersionAtLeast11And2" or something like this.